shelley/chain-and-ledger/formal-spec/Properties.md

```diff
@@ -1,3 +1,34 @@
+# Ledger and Epoch State Validity
+
+We only care that the properties below are satisfied for _valid_ ledger states, and
+more generally, valid _epoch_ states. Checking things for invalid states should
+not be performed. For this reason, we must have boolean functions that
+evaluate to `True` for valid states. As the STS rule system we have defined
+is deterministic, all valid states can be reached using the transitions in the system,
+and the only states that are valid are those that can be described by a sequence
+of rule applications. Thus, a high-level definition of such a validity function
+would be:
+
+* `validState(s) = True` when `s` is either a valid initial state, or there is
+some sequence of transition rules that can be applied, starting from an
+initial state, to reach the state `s`
+- we say `s` is a valid state
```

Comment:
I think this is a good definition, but then what is a valid initial state ? In the spec we have a function to construct a Shelley initial state from a Byron state. Maybe we say, any state that can be produced by this mapping is a valid state? A Byron state is much simpler, so we could define a valid Byron state as some non-conflicting set of constraints and get a valid Shelley initial state from there?

Reply:
i am changing the spec right now so this discussion will live in the Properties section of the LaTeX doc (i think Jared was hoping to make this md file more high-level with no specific definitions). I removed (temporarily, until we agree on the correct constructive definition) the `validLedgerState` definition using the LEDGER transition, and replaced it with just the type: 

`validLedgerState ∈ LEnv → LState → Bool`

Also - I think this map does need to take the environment as a parameter?

Reply:
As for the Byron state - yeah. I will say that ^^